internal/controllers/machine/machine_controller.go

```diff
@@ -687,6 +685,13 @@ func (r *Reconciler) shouldAdopt(m *clusterv1.Machine) bool {
 }
 
 func (r *Reconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	if cluster.GetConditions() == nil || conditions.IsFalse(cluster, clusterv1.ControlPlaneInitializedCondition) {
```

Comment:
```suggestion
	if !conditions.IsTrue(cluster, clusterv1.ControlPlaneInitializedCondition) {
```
should include both conditions

Reply:
fixed, thanks!